src/lib.rs

```diff
@@ -598,7 +601,8 @@ impl IntoIter {
         // Make room for another open file descriptor if we've hit the max.
         if self.stack_list.len() - self.oldest_opened == self.opts.max_open {
             self.stack_list[self.oldest_opened].close();
-            self.oldest_opened = self.oldest_opened.checked_add(1).unwrap();
+            // Unwrap is safe here as long as we assume that `self.opts.max_open` is never 0
```

Comment:
Hmm. I don't think this is right. While we do actually *guarantee* that `self.ops.max_open` is greater than `0`, that doesn't seem to have anything to do with this unwrap. In particular, this unwrap is just making sure that we don't overflow the `oldest_opened` state, which is a `usize`. It's not technically a bug in walkdir if this trips, but it's more like an error case that really can't happen in practice. In particular, it would require descending into a directory that has a depth greater than the maximum size of a `usize` on the current platform.

Arguably, one might want to turn this into a real error, but it doesn't seem worth it to me. I think we should just add a comment that basically says what I said above and call it a day. What do you think?